puffin/src/lib.rs

```diff
@@ -629,17 +629,47 @@ macro_rules! current_function_name {
 
 #[doc(hidden)]
 #[inline]
-pub fn clean_function_name(name: &str) -> &str {
+pub fn clean_function_name(name: &str) -> String {
```

Comment:
nit: this shouldn't make an active difference, though is a bit cleaner as we want to only operate on static strings for the file & function names.

```suggestion
pub fn clean_function_name(name: &'static str) -> String {
```